src/algorithms/preprune.rs

```diff
@@ -0,0 +1,116 @@
+use crate::{Correctness, Guess, Guesser, DICTIONARY};
+use once_cell::sync::OnceCell;
+use std::borrow::Cow;
+
+static INITIAL: OnceCell<Vec<(&'static str, usize)>> = OnceCell::new();
+static PATTERNS: OnceCell<Vec<[Correctness; 5]>> = OnceCell::new();
+
+pub struct PrePrune {
```

Comment:
I think we should just fix `Prune` instead of adding this as a separate implementation. I think we should also apply that fix diff to the other implementations that build on `Prune` (like `cutoff`).